test/src/main/java/dev/drewhamilton/androiddatetimeformatters/test/TimeSettingTest.kt

```diff
@@ -58,10 +48,50 @@ abstract class TimeSettingTest {
         }
 
     protected var testLocale: Locale
-        get() = testContext.resources.configuration.locale
+        get() = ConfigurationCompat.getLocales(testContext.resources.configuration)[0]
         set(value) {
-            testContext.resources.configuration.locale = value
+            setLocales(LocaleListCompat.create(value))
+        }
+
+    @Before fun cacheOriginalTimeSetting() {
+        originalTimeSetting = systemTimeSetting
+        // Ignore the test if the device won't allow resetting to the original hour setting:
+        ignoreIfThrowing<IllegalArgumentException> {
+            forceRestoreSystemTimeSetting()
+        }
+        canResetSystemTimeSetting = true
+        Log.d(TAG, "cached original setting: $originalTimeSetting")
+    }
+
+    @After fun restoreTimeSetting() {
+        if (canResetSystemTimeSetting) {
+            forceRestoreSystemTimeSetting()
+            Log.d(TAG, "Reset original setting: $originalTimeSetting")
         }
+    }
+
+    private fun forceRestoreSystemTimeSetting() {
+        systemTimeSetting = originalTimeSetting
+    }
+
+    @Before fun cacheOriginalLocales() {
+        originalLocales = ConfigurationCompat.getLocales(testContext.resources.configuration)
+    }
+
+    @After fun restoreLocales() {
```

Comment:
Check if I actually need this

Reply:
This is not needed on APIs after Jelly Bean, but may be needed on Jelly Bean and earlier devices. I can't actually run tests on such devices right now, so I'll leave this here just in case.